trackers/core/reid/callbacks.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Optional
+
+import wandb
+from torch.utils.tensorboard import SummaryWriter
+
+
+class BaseCallback:
+    def on_train_batch_start(self, logs: dict, idx: int):
+        pass
+
+    def on_train_batch_end(self, logs: dict, idx: int):
+        pass
+
+    def on_validation_batch_start(self, logs: dict, idx: int):
+        pass
+
+    def on_validation_batch_end(self, logs: dict, idx: int):
+        pass
```

Comment:
Do we want to log on every batch or on every epoch?

Reply:
I like to log after every batch to get the raw insights.
If the logs per batch look too dense, we can add events to the callback called `on_epoch_end` where the accumulated metrics across the batch are logged.

Reply:
Hi @SkalskiP, I updated the metric logger callbacks in the following manner:
- Only train loss is logged every batch step under the key `batch/train/loss`
- Train loss is also logged after every epoch under the key `train/loss`
- Validation loss is logged only after every epoch under the key `validation/loss`

Would love to know your thoughts on this.